src/main/java/de/tum/in/www1/artemis/web/rest/ComplaintResponseResource.java

```diff
@@ -62,26 +59,66 @@ public ComplaintResponseResource(ComplaintResponseRepository complaintResponseRe
     }
 
     /**
-     * POST /complaint-responses: create a new complaint response
+     * POST /complaint-responses/complaint/:complaintId/create-lock: locks the complaint by creating an empty complaint response
      *
-     * @param complaintResponse the complaint response to create
-     * @return the ResponseEntity with status 201 (Created) and with body the new complaint response
-     * @throws URISyntaxException if the Location URI syntax is incorrect
+     * @param complaintId - id of the complaint to lock
+     * @return the ResponseEntity with status 201 (Created) and with body the empty complaint response
      */
-    @PostMapping("/complaint-responses")
+    @PostMapping("/complaint-responses/complaint/{complaintId}/create-lock")
     @PreAuthorize("hasAnyRole('TA', 'INSTRUCTOR', 'ADMIN')")
-    public ResponseEntity<ComplaintResponse> createComplaintResponse(@RequestBody ComplaintResponse complaintResponse) throws URISyntaxException {
-        log.debug("REST request to save ComplaintResponse: {}", complaintResponse);
-        ComplaintResponse savedComplaintResponse = complaintResponseService.createComplaintResponse(complaintResponse);
-
-        // To build correct creation alert on the client we must check which type is the complaint to apply correct i18n key.
-        String entityName = complaintResponse.getComplaint().getComplaintType() == ComplaintType.MORE_FEEDBACK ? MORE_FEEDBACK_RESPONSE_ENITY_NAME : ENTITY_NAME;
+    public ResponseEntity<ComplaintResponse> lockComplaint(@PathVariable long complaintId) {
+        log.debug("REST request to create empty complaint response for complaint with id: {}", complaintId);
```

Comment:
seems that permission checks are missing here

Reply:
Done in the service. See:
private boolean isUserAuthorizedToRespondToComplaint(StudentParticipation participation, Complaint